### PR TITLE
Add confs for nikshay private api integration on softlayer

### PR DESCRIPTION
**Pull request by mkangia**

### CODE REVIEW

File: ansible/vars/softlayer/softlayer_public.yml [hunk 1]
```diff
@@ -270,3 +270,5 @@ localsettings:
   USE_PARTITIONED_DATABASE: True
   WS4REDIS_CONNECTION_HOST: "{{ groups.redis.0 }}"
   XFORMS_PLAYER_URL: "http://{{ groups.touchforms.0 }}:4444/"
+  ENIKSHAY_PRIVATE_API_PASSWORD: {'BI': 'ppia-bi21', 'GU': 'ppia-gu21', 'MH': 'ppia-mh14'}
```
Comment (benrudolph) [inline ansible/vars/softlayer/softlayer_public.yml]: should this be `ENIKSHAY_PRIVATE_API_USERS`?

Comment (benrudolph) [inline ansible/vars/softlayer/softlayer_public.yml]: also should this be public? `PRIVATE` implies not but maybe that's for something else

File: ansible/roles/commcarehq/templates/localsettings.py.j2 [hunk 2]
```diff
@@ -679,3 +679,11 @@ ENABLE_PASSWORD_HASHING = True
 {% if localsettings.get('ENABLE_USED_PASSWORDS_CHECK') %}
 ENABLE_USED_PASSWORDS_CHECK = True
 {% endif %}
+
+{% if localsettings.get('ENIKSHAY_PRIVATE_API_USERS') %}
+ENIKSHAY_PRIVATE_API_USERS = True
```
Comment (benrudolph) [inline ansible/roles/commcarehq/templates/localsettings.py.j2]: this should probably be `ENIKSHAY_PRIVATE_API_USERS = localsettings.get('ENIKSHAY_PRIVATE_API_USERS')` unless i'm misunderstanding (same with the below)

File: ansible/vars/softlayer/softlayer_public.yml (hunk 1 above)
Comment (mkangia) [inline ansible/vars/softlayer/softlayer_public.yml]: PRIVATE is for PRIVATE API. There are two different integration, one private and one public.

File: ansible/roles/commcarehq/templates/localsettings.py.j2 (hunk 2 above)
Comment (mkangia) [inline ansible/roles/commcarehq/templates/localsettings.py.j2]: Sorry for the silly mistake there.

File: ansible/vars/softlayer/softlayer_public.yml [hunk 3]
```diff
@@ -270,3 +270,5 @@ localsettings:
   USE_PARTITIONED_DATABASE: True
   WS4REDIS_CONNECTION_HOST: "{{ groups.redis.0 }}"
   XFORMS_PLAYER_URL: "http://{{ groups.touchforms.0 }}:4444/"
+  ENIKSHAY_PRIVATE_API_USERS: {'BI': 'ppia-bi21', 'GU': 'ppia-gu21', 'MH': 'ppia-mh14'}
```
Comment (benrudolph) [inline ansible/vars/softlayer/softlayer_public.yml]: 👍 i'm not sure that `yml` will take this format, otherwise looks good

Comment (benrudolph) [inline ansible/vars/softlayer/softlayer_public.yml]: (maybe do a test run to find out?)

Comment (mkangia) [inline ansible/vars/softlayer/softlayer_public.yml]: there is a similar declaration in the same file.
https://github.com/dimagi/commcarehq-ansible/pull/779/files#diff-55d98789f0f1ed501c2fc3bf8a9a0c9bR176

**Pull request merged**